backend/src/main/java/com/example/user/UserService.java

```diff
@@ -98,5 +98,15 @@ public void changePassword(UserController.ChangePasswordDTO changePasswordDTO) {
         user.get().setPassword(changePasswordDTO.getNewPassword());
         userRepository.save(user.get());
     }
+
+    public void changeEmail(UserController.ChangeEmailDTO changeEmailDTO) {
+        final var nickname = changeEmailDTO.getNickname();
+        Optional<User> user = userRepository.findByName(nickname);
+        if (user.isEmpty()) {
```

Comment:
`final var user = userRepository.findById(id).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Could not find an user with this name"));`